### ISSUE TITLE
How do we successfully evaluate our model?

### ISSUE THREAD

**Issue by ZER-0-NE** [axiom-data-science/OrcaCNN-data#1]
I believe we cannot wholly depend on accuracy metric(or do we?) to evaluate our model since there might be some erroneous case where the prediction is actually on some noise rather than on the killer whale calls. How do we make sure that we have actually predicted the presence of a killer whale call.?

One thing that comes to my mind is to clip off those frequency ranges which does not lie in the range for killer whale calls.
Am I correct in my reasoning? @yosoyjay

**Comment by yosoyjay** [comment]
For the time being, we'll just have to use the data that we have at hand.  It will be at least 5 weeks before we can get additional data.

Further on, we'll be able to develop a validation set that we can use in addition to the training/test split I'm assuming that you're using.

We'll also have domain expertise from the co-mentor who will be able to help with that.  He's the expert on these calls and such, so help develop a plan to provide verification.

**Comment by ZER-0-NE** [comment]
That would be perfect!
@yosoyjay I've implemented some more models (CNN, CNN+RNN). The RNN includes LSTM cells with and without data augmentation.
The accuracies reached max after 2-3 epochs in the CNN + RNN models. I believe them to be the best models so far.
I've attached screenshots for the same:

**CNN Model**
![screenshot from 2019-03-07 11-57-32](https://user-images.githubusercontent.com/19359908/53936608-63029680-40d0-11e9-81f2-f6059550c675.png)


**CNN+RNN (LSTM cells)**
![screenshot from 2019-03-07 11-57-56](https://user-images.githubusercontent.com/19359908/53936631-84fc1900-40d0-11e9-84a9-70e8cddaf073.png)


**CNN + RNN(LSTM) with data augmentation**
![screenshot from 2019-03-07 11-58-08](https://user-images.githubusercontent.com/19359908/53936651-9513f880-40d0-11e9-89f3-5d630ad2753b.png)


I stopped some of the models earlier when they achieved max accuracy.

**Comment by ZER-0-NE** [comment]
What I've gathered from this trial and error:

- Only single and double conv layers do not work well for time series data, like audio here.
- A combination of CNN + RNN seems to work well.
- Data Augmentation (adding noise) seems to not make much significant difference (which is kind of unexpected).

**Comment by yosoyjay** [comment]
Interesting and promising results!  It's not terribly surprising that the RNN help out since there is a time-dimension in the problem.

How exactly that you apply the data augmentation?  I could imagine a scenario where if it were not carefully applied, the network would quickly filter it out and it would not affect results.

**Comment by ZER-0-NE** [comment]
> How exactly that you apply the data augmentation? I could imagine a scenario where if it were not carefully applied, the network would quickly filter it out and it would not affect results.

I researched and found two techniques initially, adding noise and shifting. As a start, I tried applied randomly generated noise to data using numpy. 
I believe it must be the case you mentioned.
